api/src/main/java/org/openmrs/validator/OrderFrequencyValidator.java

```diff
@@ -61,7 +61,7 @@ public void validate(Object obj, Errors errors) {
 			
 			Concept concept = orderFrequency.getConcept();
 			if (concept != null) {
-				if (!ConceptClass.FREQUENCY_UUID.equals(concept.getConceptClass().getUuid())) {
+				if (!ConceptClass.FREQUENCY_UUID.equals(concept.getConceptClass().getUuid().trim())) {
```

Comment:
Aren't we going to have to do this in many places? Did you evaluate other options? Like switching the file size to 36?

Reply:
@dkayiwa , yes and as I am testing and getting validation errors, I am making the changes. One another ticket I raised today holds a similar problem. 

Making uuid columns char(36) or varchar(38) did come in my mind, but isn't there any reason that why in OpenMRS we have set UUID columns as char(38) everywhere?

Reply:
https://issues.openmrs.org/browse/TRUNK-1283

Reply:
Can we write the custom liquibase class that automatically maps to char(36) for PostgreSQL?

Reply:
Let me look into it. I know something similar has been done for boolean conversion to tinyint(1) for MySQL.

Reply:
I'd actually think that we'd want to just change all UUID columns to 36 characters. The likelihood that we need to contend with Microsoft-generated UUIDs seems to me exceedingly small (and in those cases, we should just be removing the curly-braces anyways).

Reply:
@ibacher , setting UUID to 36 chars gives an error : 

`
Caused by: java.sql.BatchUpdateException: Batch entry 0 insert into appointmentscheduling_appointment_type (name, description, duration, confidential, visit_type_id, uuid, date_created, date_changed, date_retired, retired, retire_reason, creator, changed_by, retired_by, appointment_type_id) values ('Urology (New Patient)', NULL, 20, 'FALSE', NULL, '4d85dda4-c437-11e4-a470-82b0ea87e2d8e', '2020-07-27 15:33:49+05:30', NULL, NULL, 'FALSE', NULL, 2, NULL, NULL, 22) was aborted: ERROR: value too long for type character(36)  Call getNextException to see other errors in the batch.
`

If you see length of '4d85dda4-c437-11e4-a470-82b0ea87e2d8e' is 37, even though there are no curly braces. Thus I shifted to using VARCHAR(38) instead. Link to logs: https://pastebin.com/3i3L2N7a

Reply:
@Am-Coder Do examples like that exist outside of the referencedemodata module? If not, shouldn't we just fix that module?

Reply:
@ibacher , Let me check after correcting this uuid if I get error refering to uuid in other modules while initializing Reference Application.

Reply:
@ibacher , I just corrected it [here](https://github.com/openmrs/openmrs-module-referencedemodata/blob/abbce3805f7c76c57a210cb8595352cca7812f34/api/src/main/java/org/openmrs/module/referencedemodata/bundle/AppointmentMetadata.java#L46) and there is no other error in any other module or reference demo data module related to uuid being more then 36 characters. But for correction I just removed the last extra character. So, we are good to go with char(36) for PostgreSQL after this correction.

@dkayiwa , what is your say on this to keep uuid to char(36) or varchar(38) and whether this be done for MySQL as well ?

Reply:
Doing a uuid of 36 characters makes more sense.

Reply:
Made the changes to convert char(38) to char(36) for PostgreSQL.